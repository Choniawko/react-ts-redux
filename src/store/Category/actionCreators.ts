import ActionCreator from "../actionCreator"

export default {
    getCategories: ActionCreator("CATEGORY_GET_CATEGORIES"),
    updateCategories: ActionCreator("CATEGORY_UPDATE_CATEGORIES"),
    setError: ActionCreator("CATEGORY_SET_ERROR")
}
