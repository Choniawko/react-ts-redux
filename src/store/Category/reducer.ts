import { ActionType, getType } from "typesafe-actions";
import actionsCreators from "./actionCreators";

export type Category = { id: number; name: string };
export type State = {
  categories: ReadonlyArray<Category>;
  count: number;
  previos: number | null;
  next: number | null;
};
export type Action = ActionType<typeof actionsCreators>;

export const INITIAL_STATE: State = {
  categories: [],
  count: 0,
  previos: null,
  next: null
};

export default (state: State = INITIAL_STATE, { type, payload }: Action) => {
  switch (type) {
    case actionsCreators.updateCategories.type:
      return {
        ...state,
        categories: payload
      };
    default:
      return state;
  }
};
