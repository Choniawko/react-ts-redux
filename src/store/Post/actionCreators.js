import ActionCreator from "../actionCreator"

export default {
    getPosts: ActionCreator("POST_GET_POSTS"),
    updatePosts: ActionCreator("POST_UPDATE_POSTS"),
    setError: ActionCreator("POST_SET_ERROR")
}
