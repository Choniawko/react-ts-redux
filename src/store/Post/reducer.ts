import { ActionType } from "typesafe-actions";
import actionsCreators from "./actionCreators";
import { Category } from "../Category/reducer";

export type Post = {
  id: number;
  title: string;
  category: Category;
  description: string;
  active: boolean;
};
export type State = {
  posts: ReadonlyArray<Post>;
  count: number;
  previos: number | null;
  next: number | null;
};
type Action = ActionType<typeof actionsCreators>;

export const INITIAL_STATE: State = {
  posts: [],
  count: 0,
  previos: null,
  next: null
};

export default (state: State = INITIAL_STATE, { type, payload }: Action) => {
  switch (type) {
    case actionsCreators.updatePosts.type:
      return {
        ...state,
        posts: payload
      };
    default:
      return state;
  }
};
