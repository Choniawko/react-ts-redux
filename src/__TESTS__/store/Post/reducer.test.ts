import reducer, { INITIAL_STATE } from "../../../store/Post/reducer";
import actionCreators from "../../../store/Post/actionCreators";

const testPost = {
  id: 1,
  title: "Post #1",
  category: "Category #1",
  description: "Description #1",
  active: true
};

describe("Post reducer", () => {
  it("updatePosts action", () => {
    expect(
      reducer(INITIAL_STATE, {
        type: actionCreators.updatePosts.type,
        payload: [testPost]
      })
    ).toMatchSnapshot();
  });
});
