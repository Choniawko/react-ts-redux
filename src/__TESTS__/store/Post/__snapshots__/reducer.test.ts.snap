// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post reducer updatePosts action 1`] = `
Object {
  "count": 0,
  "next": null,
  "posts": Array [
    Object {
      "active": true,
      "category": "Category #1",
      "description": "Description #1",
      "id": 1,
      "title": "Post #1",
    },
  ],
  "previos": null,
}
`;
